### PR TITLE
feat: add providers for details

### PR DESCRIPTION
**Pull request by Primdahl26**
# Description

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: frontend/src/lib/components/media_card.svelte [hunk 1]
```diff
@@ -11,6 +11,9 @@
   export let mediaIndex: number
   export let media: Hit
 
+  // console.log(media)
+  // console.log(providerAmounts)
```
[the remark this answers is not in the document]
Comment (Primdahl26) [reply frontend/src/lib/components/media_card.svelte]: Done

File: frontend/src/lib/components/details/person_media.svelte [hunk 2]
```diff
@@ -1,40 +1,54 @@
...
-  } from "../../utils"
-  import type { Media } from "../../types"
-  import { IMG_W342 } from "$lib/variables"
-  import { onMount } from "svelte"
+  } from "$lib/utils"
+  import type { Media } from "$lib/types"
+  import { PYTHON_API } from "$lib/variables"
+  import { currentCountry } from "$lib/stores/country"
+  import type { Meilisearch, Hit } from "$lib/generated"
+  import Spinner from "../loading/spinner.svelte"
```
Comment (AndreasPB) [inline frontend/src/lib/components/details/person_media.svelte:12]: $lib

File: frontend/src/lib/components/details/person_media.svelte [hunk 3]
```diff
@@ -1,40 +1,54 @@
...
+    return json
+  }
+
+  let refreshLookupMedia = lookupMedia
 
-  const loadMoreData = async ({ detail: { loaded } }) => {
-    currentMediaAmount += mediaStartAmount
-    loaded()
+  $: if ($currentCountry) {
+    refreshLookupMedia = lookupMedia
```
Comment (AndreasPB) [inline frontend/src/lib/components/details/person_media.svelte:51]: Might be more descriptive of the "hack"

```suggestion
    refreshableLookupMedia = lookupMedia
```

File: frontend/src/lib/components/details/person_media.svelte [hunk 4]
```diff
@@ -44,31 +58,16 @@
...
-      </a>
-    {/each}
-  </div>
-  {#if media.length > currentMediaAmount}
-    <InfiniteLoading on:infinite={loadMoreData} />
-  {:else}
-    <p class="text-center italic">Showing {media.length} results</p>
-  {/if}
+  {#await refreshLookupMedia()}
+    <Spinner timeout={false} />
```
Comment (AndreasPB) [inline frontend/src/lib/components/details/person_media.svelte:62]: This is fine for now, but should we make an issue about the skeleton cards?

File: frontend/src/lib/components/details/recommendations.svelte [hunk 5]
```diff
@@ -1,17 +1,49 @@
...
   import { Navigation, Lazy } from "swiper"
+  import MediaCard from "$lib/components/media_card.svelte"
   import "swiper/css"
   import "swiper/css/free-mode"
-  import { IMG_W342 } from "../../variables"
+  import { PYTHON_API } from "$lib/variables"
   import { onMount } from "svelte"
+  import { currentCountry } from "$lib/stores/country"
+  import type { Meilisearch, Hit } from "$lib/generated"
+  import Spinner from "../loading/spinner.svelte"
```
Comment (AndreasPB) [inline frontend/src/lib/components/details/recommendations.svelte:12]: $lib

File: frontend/src/lib/components/details/person_media.svelte [hunk 6]
```diff
@@ -1,40 +1,54 @@
...
+      `${PYTHON_API}/media?c=${$currentCountry}&limit=${ids.length}&ids=${ids.join(
+        "&ids="
+      )}`
+    )
+    const json: Meilisearch = await res.json()
+
+    hitProviderAmounts(json.hits)
+
+    return json
+  }
```
Comment (AndreasPB) [inline frontend/src/lib/components/details/person_media.svelte:46]: There's 3 of these, it should be put in `util.ts`. Move the imperative media.map/recommendations.map to a parameter(functional code bby)

File: frontend/src/lib/components/details/person_media.svelte (hunk 4 above)
Comment (Primdahl26) [reply frontend/src/lib/components/details/person_media.svelte:62]: #385

File: frontend/src/lib/components/details/recommendations.svelte [hunk 7]
```diff
@@ -1,17 +1,49 @@
...
+        }
+        if ("free" in hit.providers.results[$currentCountry]) {
+          combinedAmount += hit.providers.results[$currentCountry]["free"].length
+        }
+      }
+      providerAmounts.push(combinedAmount)
+    })
+  }
+
+  const lookupRecommendations = async (): Promise<Meilisearch> => {
```
Comment (AndreasPB) [inline frontend/src/lib/components/details/recommendations.svelte:36]: Same as lookupMedia

File: frontend/src/lib/components/details/person_media.svelte (hunk 2 above)
Comment (Primdahl26) [reply frontend/src/lib/components/details/person_media.svelte]: Done

File: frontend/src/lib/components/details/person_media.svelte (hunk 3 above)
Comment (Primdahl26) [reply frontend/src/lib/components/details/person_media.svelte]: Deleted it after making it functional

File: frontend/src/lib/components/details/recommendations.svelte (hunk 5 above)
Comment (Primdahl26) [reply frontend/src/lib/components/details/recommendations.svelte]: Done

File: frontend/src/lib/components/details/person_media.svelte (hunk 6 above)
Comment (Primdahl26) [reply frontend/src/lib/components/details/person_media.svelte]: Done

File: frontend/src/lib/components/details/recommendations.svelte (hunk 7 above)
Comment (Primdahl26) [reply frontend/src/lib/components/details/recommendations.svelte]: Done

**Review by AndreasPB — APPROVED** [review]

**Pull request merged**